src/families/cosmos/types.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+
+import type { BigNumber } from "bignumber.js";
+import type {
+  TransactionCommon,
+  TransactionCommonRaw
+} from "../../types/transaction";
+
+export type CoreStatics = {};
+export type CoreAccountSpecifics = {};
+export type CoreOperationSpecifics = {};
+export type CoreCurrencySpecifics = {};
+
+export type CosmosDelegation = {|
+  validatorAddress: string,
+  amount: BigNumber,
+  pendingRewards: BigNumber,
+  isActive: boolean, // TBD is there more status case? if so, switch to status field with an enum string.
+|};
+
+export type CosmosRedelegation = {|
+  sourceValidatorAddress: string,
+  dstValidatorAddress: string,
+  completionTime: Date,
+  amount: BigNumber,
+|};
+
+export type CosmosUnbounding = {|
+  validatorAddress: string,
+  completionTime: Date,
+  amount: BigNumber,
+|};
+
+export type CosmosResources = {|
+  delegations: CosmosDelegation[],
+  redelegations: CosmosRedelegation[],
+  unbounding: CosmosUnbounding[],
+  delegatedBalance: BigNumber,
+  pendingRewardsBalance: BigNumber, 
+  unboundingBalance: BigNumber,
+  withdrawAddress: string,
+|};
+
+export type CosmosDelegationRaw = {|
+  validatorAddress: string,
+  amount: string,
+  pendingRewards: string,
+  isActive: boolean,
+|};
+
+export type CosmosRedelegationRaw = {|
+  sourceValidatorAddress: string,
+  dstValidatorAddress: string,
+  completionTime: string,
+  amount: string,
+|};
+
+export type CosmosUnboundingRaw = {|
+  validatorAddress: string,
+  completionTime: string,
+  amount: string,
+|};
+
+export type CosmosResourcesRaw = {|
+  delegations: CosmosDelegationRaw[],
+  redelegations: CosmosRedelegationRaw[],
+  unbounding: CosmosUnboundingRaw[],
+  delegatedBalance: string,
+  pendingRewardsBalance: string, 
+  unboundingBalance: string,
+  withdrawAddress: string,
+|};
+
+// NB this must be serializable (no Date, no BigNumber)
+export type CosmosValidatorItem = {|
+  validatorAddress: string,
+  name: string,
+  votingPower: number, // value from 0.0 to 1.0 (normalized percentage)
+  commission: number, // value from 0.0 to 1.0 (normalized percentage)
+|};
+
+// by convention preload would return a Promise of CosmosPreloadData
+export type CosmosPreloadData = {
+  validators: CosmosValidatorItem[]
+};
+
+export type CosmosOperationMode =
+  | "send"
+  | "delegate"
+  | "undelegate"
+  | "redelegate"
+  | "claimReward"
+  | "claimAllRewards";
+
+export type NetworkInfo = {|
+  family: "cosmos",
+  fees: BigNumber
+|};
+
+export type NetworkInfoRaw = {|
+  family: "cosmos",
+  fees: string
+|};
+
+// TODO suggest to rename to CosmosValidatorInput?
+export type CosmosValidator = {
+  address: string,
+  amount: BigNumber
+};
+
+export type CosmosValidatorRaw = {
+  address: string,
+  amount: string
+};
+
+export type Transaction = {|
```

Comment:
In Cosmos, a Transaction is a list of messages (each with its own `mode`), wrapped into `memo`, `fees`, `accountNumber`, `sequence`.... elements.

So storing a Transaction with this model would only work for uni-message transactions, and that would not work for multiple message transactions like [this one](https://hubble.figment.network/cosmos/chains/cosmoshub-3/transactions/0DBFC4E8E9E5A64C2C9B5EAAAA0422D99A61CFC5354E15002A061E91200DC2D6) (WithdrawRewards + WithdrawCommission messages ; don't mind these message types, WithdrawCommission is Validator-specific)

Reply:
i don't understand your feedback. this is exactly from your current ongoing implementation. cc @henri-ly .
A Transaction here is a UX concept which holds all the state the UI need (user inputs, other volatile infos,..)

Reply:
Oh, alright. I haven't been looking too much the code in `live-common` for a while. I suppose we manually unwrap operations then, sorry for the noise